### PR TITLE
Implements TokenManager with RSA JWT signing

### PR DESCRIPTION
**Pull request by bbengfort**
Creates a tokens package with a TokenManager that is used to generate and verify JWT tokens using RSA keys from the configuration defined in SC-563. Upgrades GDS to use the new golang-jwt v4 package.

TODO:
- [x] Implement claims for GDS
- [x] Generate signed keys using TokenManager
- [x] Ensure kid is in the token header
- [x] Verify keys using TokenManager

Fixes SC-564

Blocked for review by #93

### CODE REVIEW

**Comment by bbengfort** [comment]
@rebeccabilbro this is ready for review after #93

File: pkg/gds/tokens/tokens_test.go [hunk 1]
```diff
@@ -0,0 +1,193 @@
...
+	require.Len(keys, 2)
+	require.Equal("1yAxs5vPqCrg433fPrFENevvzen", tm.CurrentKey().String())
+
+	// Create an access token from simple claims
+	creds := map[string]interface{}{
+		"hd":      "rotational.io",
+		"email":   "kate@rotational.io",
+		"name":    "Kate Holland",
+		"picture": "https://foo.googleusercontent.com/test!/Aoh14gJceTrUA",
+	}
```
Comment (rebeccabilbro) [inline pkg/gds/tokens/tokens_test.go:38]: This example helps!

File: pkg/gds/tokens/tokens_test.go [hunk 2]
```diff
@@ -0,0 +1,193 @@
...
+	require.NoError(err, "could not validate access token")
+
+	// Validate the refresh token (should be invalid because of not before in the future)
+	_, err = tm.Verify(rtks)
+	require.Error(err, "refresh token is valid?")
+}
+
+// TODO: test validation of audience, issuer, and subject.
+// TODO: test time based validation (not before, issued at, expires)
+// TODO: test signed with wrong key
```
Comment (rebeccabilbro) [inline pkg/gds/tokens/tokens_test.go:101]: e.g. the prior (3 month old) key, or a bad/malicious key?

File: pkg/gds/tokens/tokens_test.go [hunk 3]
```diff
@@ -0,0 +1,193 @@
...
+		"picture": "https://foo.googleusercontent.com/test!/Aoh14gJceTrUA",
+	})
+	require.NoError(err)
+
+	tks, err := oldTM.Sign(token)
+	require.NoError(err)
+
+	// Validate token with "new token manager"
+	_, err = newTM.Verify(tks)
+	require.NoError(err)
```
Comment (rebeccabilbro) [inline pkg/gds/tokens/tokens_test.go:187]: Would we want a check for the reverse? e.g. ensure that oldTM.Verify(newTM.CreateAccessToken) returns an error? I suppose this is what you meant by test signed with wrong key in the earlier test

**Review by rebeccabilbro — APPROVED** [review]
Looks good @bbengfort &mdash; I left a few inline questions since I'm new to JWT, but I think I get the general gist!

**Comment by rebeccabilbro** [comment]
Also forgot to mention that I've updated the shortcut settings to (hopefully) register PRs and merges from branches that are named after the shortcut `sc-story_id` for *all* Rotational repos (when @tl6kk and I set it up earlier this week, we'd only enabled it for localew as an experiment). Should be working for this repo moving forward!

**Pull request merged**